### ISSUE TITLE
Redirects are always absolute no matter what `absolute` parameter value is

### ISSUE THREAD

**Issue by chrkv** [grails/grails-core#10879]
From reading documentation (http://docs.grails.org/latest/ref/Controllers/redirect.html) I got the feeling that if one doesn't pass a value or specifically passes `false` for `absolute` parameter then generated redirect would be relative. But I am always getting an absolute redirect with `serverURL` added. 

Relative redirects are required for me because we are using blue green deployment and we are constantly redirected from inactive container to active. (`serverURL` is set to active container address)

**Comment by andrewlucas** [comment]
I am also experiencing the same problem. Grails 3.2.4. This prevents SSL redirection working properly, when the origin server is behind an AWS ELB, every single form post has a redirect back to HTTP.

**Comment by ilopmar** [comment]
@chrkv @andrewlucas Just to make sure, what do you want is to return a URI in the `Location` header, right? Something like:

```
ivan@nobita ~ $ curl -v http://localhost:8080/test/action1
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /test/action1 HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 302 
< X-Application-Context: application:development
< Location: /test/action2   <------- Instead of Location: http://localhost:8080/test/action2
< Content-Length: 0
< Date: Tue, 23 Jan 2018 12:08:55 GMT
<
```

Right?

Grails always generates absolute urls for a redirect: https://github.com/grails/grails-core/blob/91734215af122b6904b046256d78d550f3076d2f/grails-web-url-mappings/src/main/groovy/grails/web/mapping/ResponseRedirector.groovy#L98
According to the RFC 2616, section 14.30: https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.30, the `Location` header is an absoluteURI:

```
The Location response-header field is used to redirect the recipient to a location other than the Request-URI for completion of the request or identification of a new resource. For 201 (Created) responses, the Location is that of the new resource which was created by the request. For 3xx responses, the location SHOULD indicate the server's preferred URI for automatic redirection to the resource. The field value consists of a single absolute URI.

       Location       = "Location" ":" absoluteURI

An example is:

       Location: http://www.w3.org/pub/WWW/People.html
```

I've also found     that RFC 7231 modifies that behavior and allows partial URIs: http://www.rfc-base.org/txt/rfc-7231.txt, section 7.1.2:

```
The "Location" header field is used in some responses to refer to a specific resource in relation to the response.  The type of relationship is defined by the combination of request method and status code semantics.

     Location = URI-reference

The field value consists of a single URI-reference.  When it has the form of a relative reference ([RFC3986], Section 4.2), the final value is computed by resolving it against the effective request URI ([RFC3986], Section 5).
For 201 (Created) responses, the Location value refers to the primary resource created by the request.  For 3xx (Redirection) responses, the Location value refers to the preferred target resource for automatically redirecting the request.

If the Location value provided in a 3xx (Redirection) response does not have a fragment component, a user agent MUST process the redirection as if the value inherits the fragment component of the URI reference used to generate the request target (i.e., the redirection inherits the original reference's fragment, if any).

For example, a GET request generated for the URI reference "http://www.example.org/~tim" might result in a 303 (See Other) response containing the header field:

     Location: /People.html#tim

which suggests that the user agent redirect to "http://www.example.org/People.html#tim"
```

So maybe Grails should support this.

The problem is that by default Grails assumes `absolute = false` so if we start generating relative URIs we may break a lot of applications.

Maybe the solution can be only generate relative URIs when `absolute = false` is explicitily set. Here https://github.com/grails/grails-core/blob/91734215af122b6904b046256d78d550f3076d2f/grails-web-url-mappings/src/main/groovy/grails/web/mapping/ResponseRedirector.groovy#L81-L85 we can check that and then pass a new param to `redirectResponse` and generate a relative URI according to that new param here: https://github.com/grails/grails-core/blob/91734215af122b6904b046256d78d550f3076d2f/grails-web-url-mappings/src/main/groovy/grails/web/mapping/ResponseRedirector.groovy#L98

@graemerocher @jameskleeh what do you think?

**Comment by ilopmar** [comment]
Closed via #10910
